src/librustdoc/doctest.rs

```diff
@@ -231,6 +277,14 @@ impl DirState {
     }
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+struct UnusedExterns {
```

Comment:
Can we at least leave a comment here that this should match the rustc definition? (And equivalently on the rustc struct?) Ideally we would deduplicate these...

Reply:
I've wondered about deduplicating the definitions but they differ by ownership, and didn't felt the benefits of deduplication justified the additional copies.

Reply:
Also note that it also appears inside cargo. A "real" solution would involve building a common crate for the json format that cargo, rustc, and rustdoc can import.

Reply:
I think we're still missing a comment here just noting that this is duplicated elsewhere. I agree on the real solution but I don't think we're there yet and it's a separate PR at least :)